test/lib/ufe/testContextOps.py

```diff
@@ -586,6 +586,160 @@ def checkItem(self, item, type, path):
+
+                if (selectedObjCount == 1):
+                    selection = ufe.GlobalSelection.get()
+                    self.assertEqual(len(selection), selectedObjCount)
+                    self.assertEqual(selection.front().nodeName(), "{0}1".format(matType))
+                    self.assertEqual(selection.front().nodeType(), "Shader")
+
+            hier = ufe.Hierarchy.hierarchy(materialItem)
+            self.assertTrue(hier.hasChildren())
+            self.assertEqual(len(hier.children()), 1)
+            shaderItem = hier.children()[0]
+            checkItem(self, shaderItem, "Shader", "{0}/{1}1/{1}1".format(scope, matType))
+
+            materialPrim = UsdShade.Material(usdUtils.getPrimFromSceneItem(materialItem))
+            self.assertTrue(materialPrim)
+
+            surfaceOutput = materialPrim.GetSurfaceOutput(context)
+            self.assertTrue(surfaceOutput)
+
+            sourceInfos, invalidPaths = surfaceOutput.GetConnectedSources()
+            self.assertEqual(len(sourceInfos), 1)
+            self.assertEqual(sourceInfos[0].source.GetPath(), Sdf.Path("{0}/{1}1/{1}1".format(scope, matType)))
+            self.assertEqual(sourceInfos[0].sourceName, shaderOutputName)
+
+        checkMaterial(self, rootHier, 4, 1, 1, 0, "UsdPreviewSurface", "", "surface")
+
+        # Re-select our multiple objects so that we can repeat the test with another material.
+        ufe.GlobalSelection.get().append(cubeItem)
+        ufe.GlobalSelection.get().append(sphereItem)
+
+        cmdSS = contextOps.doOpCmd(['Assign New Material', 'MaterialX', 'ND_standard_surface_surfaceshader'])
+        self.assertIsNotNone(cmdSS)
+        ufeCmd.execute(cmdSS)
+
+        checkMaterial(self, rootHier, 4, 2, 1, 0, "UsdPreviewSurface", "", "surface")
+        checkMaterial(self, rootHier, 4, 2, 1, 1, "standard_surface", "mtlx", "out")
+
+        cmds.undo()
+
+        checkMaterial(self, rootHier, 4, 1, 3, 0, "UsdPreviewSurface", "", "surface")
+
+        cmds.redo()
+
+        checkMaterial(self, rootHier, 4, 2, 1, 0, "UsdPreviewSurface", "", "surface")
+        checkMaterial(self, rootHier, 4, 2, 1, 1, "standard_surface", "mtlx", "out")
+
+        cmds.undo()
+        cmds.undo()
+
+        # Not even a mtl scope:
+        self.assertEqual(len(rootHier.children()), 3)
+        self.assertFalse(capsulePrim.HasAPI(UsdShade.MaterialBindingAPI))
+        self.assertFalse(cubePrim.HasAPI(UsdShade.MaterialBindingAPI))
+        self.assertFalse(spherePrim.HasAPI(UsdShade.MaterialBindingAPI))
+
+        cmds.redo()
+
+        checkMaterial(self, rootHier, 4, 1, 1, 0, "UsdPreviewSurface", "", "surface")
+
+        os.putenv("MAYAUSD_MATERIALS_SCOPE_NAME", "test_scope")
+        ufeCmd.execute(cmdSS)
+        checkMaterial(self, rootHier, 5, 1, 1, 0, "standard_surface", "mtlx", "out", "/test_scope")
+
+        # Clear the envvar, otherwise we're affecting the results of following tests.
+        os.putenv("MAYAUSD_MATERIALS_SCOPE_NAME", "")
```

Comment:
I think we should have a way to make sure this is automatically restored using a context manager object. Look at `TemporaryDirectory`. You can create one that will restore an env var. The input would be the env var and the new value. The __init__ will get the current value, store it and set the env var to input and the __exit__ will restore to stored value. Then you can wrap this section of code in a `with ...` statement.